cub/test/CMakeLists.txt

```diff
@@ -269,6 +269,7 @@ function(cub_add_test target_name_var test_name test_src cub_target launcher_id)
       thrust_fix_clang_nvcc_build_for(${test_target})
     endif()
 
+    target_compile_options(${test_target} PRIVATE -ftemplate-depth=1000) # for handling large type lists
```

Comment:
important: I'm not sure about this change. We do not control user build options. If we accidentally hit the template depth on the implementation side, now we won't notice this in our tests and could break user build. 
```suggestion
```

Reply:
Good point! I will scope it to just the iterator test then.